Sources/MongoSwift/BSON/Document+Sequence.swift

```diff
@@ -12,9 +12,7 @@ import mongoc
 extension Document: Sequence {
     /// The element type of a document: a tuple containing an individual key-value pair.
     public typealias KeyValuePair = (key: String, value: BSONValue?)
-
-    /// Returns the number of (key, value) pairs stored at the top level of this `Document`.
-    public var count: Int { return Int(bson_count_keys(self.data)) }
+    public typealias SubSequence = Document
```

Comment:
Done.

Reply:
I think this is more an implementation detail - which is important to note but I'm not sure is super helpful to users. can we make this part a `//` comment and add a `///` line below detailing what's relevant for users to know?
 "the type that will be returned from methods that return parts of documents, like `map` and `filter`", or something like that.

Reply:
So not exactly sure which parts are better for users and not, but I've opted for this:
```swift
// Since a `Document` is a recursive structure, we want to enforce the use of it as a subsequence of itself.
// instead of something like `Slice<Document>`.
/// The type that is returned from methods such as `dropFirst()` and `split()`.
```